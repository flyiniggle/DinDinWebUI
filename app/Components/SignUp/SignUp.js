import TextInput from 'UI/Forms/TextInput/TextInput';
import React from 'react';


class SignUp extends React.Component {
    constructor() {
        super();

        this.state = {
            username: undefined,
            email: undefined,
            password: undefined,
            passwordRepeat: undefined,
            usernameError: undefined,
            emailError: undefined,
            passwordError: undefined,
            passwordRepeatError: undefined
        };
    }

    signUp = () => {

    }

    update = (field, value) => {
        this.setState({
            [field]: value,
            [`${field}Error`]: undefined
        });
    }

    render() {
        return (
            <div className="row d-flex justify-content-center">
                <form className="login col-xl-3 col-lg-4 col-md-6 col-8 p-5">
                    <div className="form-group row d-flex flex-column">
                        <div className="mb-4">
                            <TextInput
                                placeholder="Username"
                                message={ this.state.usernameError }
                                value={ this.state.username }
                                onChange={ (e) => this.update('username', e.target.value) }
                            />
                        </div>

                        <div className="mb-4">
                            <TextInput
                                placeholder="Email"
                                message={ this.state.emailError }
                                value={ this.state.email }
                                onChange={ (e) => this.update('email', e.target.value) }
                            />
                        </div>

                        <div className="mb-4">
                            <TextInput
                                placeholder="Password"
                                message={ this.state.passwordError }
                                value={ this.state.password }
                                onChange={ (e) => this.update('password', e.target.value) }
                            />
                        </div>

                        <div className="mb-4">
                            <TextInput
                                placeholder="Re-enter password"
                                message={ this.state.passwordRepeatError }
                                value={ this.state.passwordRepeat }
                                onChange={ (e) => this.update('passwordRepeat', e.target.value) }
                            />
                        </div>
                        <input className="btn btn-secondary" type="button" value="Sign up!" onClick={ this.signUp } />
                    </div>
                </form>
            </div>
        );
    }
}

export default SignUp;