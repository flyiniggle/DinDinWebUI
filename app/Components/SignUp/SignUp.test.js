import React from 'react';
import { shallow } from 'enzyme';

import SignUp from './SignUp';

describe('#Components #Signup', function() {
    it('should render.', function() {
        const wrapper = shallow(<SignUp />);

        expect(wrapper.html()).toMatchSnapshot();
    });
});