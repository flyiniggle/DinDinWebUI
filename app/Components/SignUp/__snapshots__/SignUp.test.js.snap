// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Components #Signup should render. 1`] = `"<div class=\\"row d-flex justify-content-center\\"><form class=\\"login col-xl-3 col-lg-4 col-md-6 col-8 p-5\\"><div class=\\"form-group row d-flex flex-column\\"><div class=\\"mb-4\\"><div class=\\"row d-flex flex-column form-group\\"><div><input type=\\"text\\" value=\\"\\" placeholder=\\"Username\\" class=\\"form-control  \\"/></div><div class=\\"textInputFeedback position-absolute\\"></div></div></div><div class=\\"mb-4\\"><div class=\\"row d-flex flex-column form-group\\"><div><input type=\\"text\\" value=\\"\\" placeholder=\\"Email\\" class=\\"form-control  \\"/></div><div class=\\"textInputFeedback position-absolute\\"></div></div></div><div class=\\"mb-4\\"><div class=\\"row d-flex flex-column form-group\\"><div><input type=\\"text\\" value=\\"\\" placeholder=\\"Password\\" class=\\"form-control  \\"/></div><div class=\\"textInputFeedback position-absolute\\"></div></div></div><div class=\\"mb-4\\"><div class=\\"row d-flex flex-column form-group\\"><div><input type=\\"text\\" value=\\"\\" placeholder=\\"Re-enter password\\" class=\\"form-control  \\"/></div><div class=\\"textInputFeedback position-absolute\\"></div></div></div><input type=\\"button\\" class=\\"btn btn-secondary\\" value=\\"Sign up!\\"/></div></form></div>"`;
